### ISSUE TITLE
DataFrame.to_dicts fails with pyarrow 6.0.1

### ISSUE THREAD

**Issue by sid-kap** [pola-rs/polars#8018]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

`DataFrame.to_dicts` uses the function `pyarrow.Table.to_pylist` which was added in pyarrow 7.0.0.

The version bound is >= 4.0.0, so IMO we should either add a workaround that works with older versions of pyarrow, or bump the version constraint.

### Reproducible example

```python
import pyarrow
import polars as pl

pl.__version__
# '0.16.18'
pyarrow.__version__
# '6.0.1'

pl.DataFrame({
    'a': [1, 2, 3]
}).to_dicts()

# produces this error:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [8], in <cell line: 1>()
----> 1 pl.DataFrame({
      2     'a': [1, 2, 3]
      3 }).to_dicts()

File ~/miniforge-arm/envs/brr-multi-dv-zscores/lib/python3.9/site-packages/polars/dataframe/frame.py:1949, in DataFrame.to_dicts(self)
   1932 def to_dicts(self) -> list[dict[str, Any]]:
   1933     """
   1934     Convert every row to a dictionary of python-native values.
   1935 
   (...)
   1947 
   1948     """
-> 1949     return list(self.iter_rows(named=True))

File ~/miniforge-arm/envs/brr-multi-dv-zscores/lib/python3.9/site-packages/polars/dataframe/frame.py:7922, in DataFrame.iter_rows(self, named, buffer_size)
   7920 zerocopy_slice = self.slice(offset, buffer_size)
   7921 if load_pyarrow_dicts:
-> 7922     yield from zerocopy_slice.to_arrow().to_pylist()
   7923 else:
   7924     rows_chunk = zerocopy_slice.rows(named=False)

AttributeError: 'pyarrow.lib.Table' object has no attribute 'to_pylist'
```

### Expected behavior

Should return `[{'a': 1}, {'a': 2}, {'a': 3}]`

### Installed versions

<details>

```
---Version info---
Polars: 0.16.18
Index type: UInt32
Platform: macOS-13.0-arm64-arm-64bit
Python: 3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:38:11) 
[Clang 14.0.6 ]
---Optional dependencies---
numpy: 1.23.5
pandas: 1.5.3
pyarrow: 6.0.1
connectorx: <not installed>
deltalake: <not installed>
fsspec: 2022.7.1
matplotlib: 3.5.2
xlsx2csv: <not installed>
xlsxwriter: <not installed>
```

</details>

**Comment by alexander-beedie** [comment]
Thanks for the heads-up 👍

**Comment by alexander-beedie** [comment]
Closed by #8026.
